docs/resources.md

```diff
@@ -618,6 +619,213 @@ the container image
 [gcr.io/cloud-builders//gcs-fetcher](https://github.com/GoogleCloudPlatform/cloud-builders/tree/master/gcs-fetcher)
 does not support configuring secrets.
 
+### Cloud Event Resource
+
+The Cloud Event Resource represents a cloud event that is sent to a target `URI`
+upon completion of a `TaskRun`. The cloud event sends Tekton specific events;
+the body of the event includes the entire TaskRun spec plus status; the
+types of events defined for now are:
+
+- dev.tekton.event.task.unknown
+- dev.tekton.event.task.successful
+- dev.tekton.event.task.failed
+
+Cloud event resources are useful to notify a third party upon the completion and
+status of a `TaskRun`. In combinations with the Tekton triggers project they can
+be used to link `Task/PipelineRuns` asynchronously.
+
+To create a CloudEvent resource using the `PipelineResource` CRD:
+
+```yaml
+apiVersion: tekton.dev/v1alpha1
+kind: PipelineResource
+metadata:
+  name: event-to-sink
+spec:
+  type: cloudevent
+  params:
+  - name: targetURI
+    value: http://sink:8080
```

Comment:
i think we might want some kind of "nullsink" type, so that folks can use Tasks that take this PipelineResource, even if they don't actually have a sink?

(probably something for a follow up PR tho!)

Reply:
Oh, nice idea. 
Another approach could be to support - in general - optional input resources - which is something that has been mentioned before, to make a Task more re-usable.

Reply:
ooo that's an interesting idea too!